pkg/local_workflows/auth_workflow_test.go

```diff
@@ -65,43 +74,133 @@ func Test_auth_token(t *testing.T) {
+		mockInvocationContext.EXPECT().GetEnhancedLogger().Return(&logger).AnyTimes()
+		err = entryPointDI(mockInvocationContext, &logger, engine, authenticator)
 		assert.NoError(t, err)
 	})
 
 	t.Run("automatically switch to token when API token is given", func(t *testing.T) {
 		config.Set(authTypeParameter, nil)
 		config.Set(ConfigurationNewAuthenticationToken, "00000000-0000-0000-0000-000000000000")
-		engine.EXPECT().InvokeWithConfig(gomock.Any(), gomock.Any())
-		err = entryPointDI(config, &logger, engine, authenticator)
+		mockInvocationContext := mocks.NewMockInvocationContext(mockCtl)
+		mockInvocationContext.EXPECT().GetConfiguration().Return(config).AnyTimes()
+		mockInvocationContext.EXPECT().GetEnhancedLogger().Return(&logger).AnyTimes()
+		engine.EXPECT().InvokeWithConfig(gomock.Any(), gomock.Any()).Return([]workflow.Data{}, nil)
+		err = entryPointDI(mockInvocationContext, &logger, engine, authenticator)
 		assert.NoError(t, err)
 	})
 }
 
+func Test_pat(t *testing.T) {
+	const (
+		testPAT               = "snyk_pat.12345678.abcdefghijklmnopqrstuvwxyz123456"
+		mockedPatEndpoint     = "https://api.snyk.io"
+		expectedAPIKeyStorage = auth.CONFIG_KEY_TOKEN
+	)
+
+	mockCtl := gomock.NewController(t)
+	defer mockCtl.Finish()
+
+	logContent := &bytes.Buffer{}
+	logger := zerolog.New(logContent)
+
+	engine := mocks.NewMockEngine(mockCtl)
+	authenticator := mocks.NewMockAuthenticator(mockCtl)
+	mockNetworkAccess := mocks.NewMockNetworkAccess(mockCtl)
+	mockHTTPClient := &http.Client{}
+
+	originalDeriveEndpointFromPAT := auth.DeriveEndpointFromPAT
+	defer func() { auth.DeriveEndpointFromPAT = originalDeriveEndpointFromPAT }()
+
+	t.Run("happy", func(t *testing.T) {
+		config := configuration.New()
+		config.Set(authTypeParameter, auth.AUTH_TYPE_PAT)
+		config.Set(ConfigurationNewAuthenticationToken, testPAT)
+		config.Set(auth.CONFIG_KEY_OAUTH_TOKEN, "some-oauth-token")
+		config.Set(configuration.AUTHENTICATION_TOKEN, "some-legacy-api-token")
+
+		mockInvocationContext := mocks.NewMockInvocationContext(mockCtl)
+		mockInvocationContext.EXPECT().GetConfiguration().Return(config).AnyTimes()
+		mockInvocationContext.EXPECT().GetEnhancedLogger().Return(&logger).AnyTimes()
+		mockInvocationContext.EXPECT().GetNetworkAccess().Return(mockNetworkAccess).Times(1)
+		mockNetworkAccess.EXPECT().GetUnauthorizedHttpClient().Return(mockHTTPClient).Times(1)
+
+		engineConfig := configuration.New()
+		engine.EXPECT().GetConfiguration().Return(engineConfig).Times(1)
+
+		auth.DeriveEndpointFromPAT = func(token string, conf configuration.Configuration, client *http.Client, patServiceUrl string) (string, error) {
+			return mockedPatEndpoint, nil
+		}
+
+		err := entryPointDI(mockInvocationContext, &logger, engine, authenticator)
+		assert.NoError(t, err)
+
+		assert.Empty(t, config.GetString(auth.CONFIG_KEY_OAUTH_TOKEN))
+		assert.Empty(t, config.GetString(configuration.AUTHENTICATION_TOKEN))
```

Comment:
Wouldn't we expect authentication to set the PAT into AUTHENTICATION_TOKEN?

Reply:
The PAT flow should resemble the legacy token flow, and in this case the AUTHENTICATION_TOKEN gets cleared to avoid it being used during the auth flow.

Reply:
To keep the PAT and legacy flows the same:
- we clear the AUTHENTICATION_TOKEN to avoid using it by mistake
- on success, we persist the new token to file, which has a different key to `AUTHENTICATION_TOKEN` (`snyk_token` vs `api`):
```
logger.Print("Set pat credentials in config")
config.PersistInStorage(auth.CONFIG_KEY_TOKEN)
config.Set(auth.CONFIG_KEY_TOKEN, pat)
```

On subsequent runs, this token in file gets set as the `AUTHENTICATION_TOKEN`.

The test checks that `CONFIG_KEY_TOKEN` is set as expected.

Reply:
Currently the `configuration.AUTHENTICATION_TOKEN` is for the legacy CLI token; we decided to use the same configuration value for PATs as they can be used interchangeably in most cases.

It's probably scope creep updating the naming, as it may have unintended side effects which need to be investigated.

As mentioned, a lot of the awkward naming is a result of how it is in the legacy CLI and the need to be backwards compatible.